### PR TITLE
Adding BBP/HBP cellular simulation packages

### PR DESCRIPTION
**Pull request by pramodk**
These packages were part of separate report repository.
We are integrating into main BBP spack repository.
 - Added mod2c
 - Added reportinglib
 - Added coreneuron
 - Added neurondamus-base (obly mod and hoc files without compilation)
 - Added coreneuron (can be built with neurodamus-base)
 - Added neurondamus (can be built with neurodamus-base and coreneuron)
 - Updated neuron packahe (will push changes upstream soon)

Todo : may be some cleanup/cosmetic changes (?)

### CODE REVIEW

File: var/spack/repos/builtin/packages/neuron/package.py [hunk 1]
```diff
@@ -44,26 +44,31 @@ class Neuron(Package):
...
+    depends_on('automake',   type='build')
+    depends_on('bison',      type='build')
+    depends_on('flex',       type='build')
     depends_on('libtool',    type='build')
     depends_on('pkg-config', type='build')
 
     depends_on('mpi',         when='+mpi')
-    depends_on('python@2.6:', when='+python')
     depends_on('ncurses',     when='~cross-compile')
+    depends_on('python@2.6:', when='+python')
```
Comment (ohm314) [inline var/spack/repos/builtin/packages/neuron/package.py:69]: I would really not base any packages on python 2.6 - unless there is a very specific need to support this version - just require 2.7+

**Review by ohm314 — APPROVED** [review]
nice :)
I only had a comment on the python version, which I would require to be newer.

File: var/spack/repos/builtin/packages/neuron/package.py (hunk 1 above)
Comment (pramodk) [reply var/spack/repos/builtin/packages/neuron/package.py:69]: We put minimum version that package can run with (which is 2.6).
But, as python recepie has latest 3.6.xx as version, Spack will always use latest python to build. So this is not a problem.

By the way, the 2.6 was default from BG-Q ;)

**Review by fouriaux — COMMENTED** [review]
Does neurodamus receipe should depends on neuron+coreneuron when variant coreneuron is used ?

**Pull request merged**